File: 2373: largestLocalValuesInAMatrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)
        limit = n-2
        res = []
        for i in range(limit):
            res.append([])
            for j in range(limit):
                currMax = grid[i][j]
                for x in range(3): 
                    for y in range(3):
                        currMax = max(currMax, grid[i+x][j+y])
                res[i].append(currMax)

        return res

File: 2373: largestLocalValuesInAMatrix.cs
public class Solution {
    public int[][] LargestLocal(int[][] grid) {
        
        int n = grid.Length;
        int[][] res = new int[n-2][];
        for(int i = 0; i < n-2; i++){
            res[i] = new int[n-2];
            for(int j = 0; j < n-2; j++){
                int curr = grid[i][j];
                for(int x = 0; x < 3; x++){
                    for(int y = 0; y < 3; y++){

                        curr = Math.Max(curr, grid[i+x][j+y]);
                    }
                }
                res[i][j] = curr;
            }
        }

        return res;
    }
}